File: Medium/Python/946. Validate Stack Sequences.py
class Solution(object):
    def validateStackSequences(self, pushed, popped):
        stack = []
        i = 0

        for n in pushed:
            stack.append(n)
            while stack and stack[-1] == popped[i]:
                stack.pop()
                i += 1

        return i == len(popped)

File: Medium/JavaScript/946. Validate Stack Sequences.js
var validateStackSequences = function(pushed, popped) {
    let stack = [];
    let i = 0;

    for(let n of pushed){
        stack.push(n);
        while(0 < stack.length && stack[stack.length - 1] === popped[i]){
            stack.pop();
            i++;
        }
    }

    return i === popped.length;
};
